features/step_definitions/crd.rb

```diff
@@ -0,0 +1,24 @@
+Given /^the #{QUOTED} #{QUOTED} CRD is recreated after scenario$/ do |name, crd|
+  ensure_admin_tagged
+  allowed_crds = ["oauth", "kubeapiserver"]
+
+  if allowed_crds.include? crd
+    @result = admin.cli_exec(:get, resource: crd, resource_name: name, o: 'yaml')
+    if @result[:success]
+      origin_crd = @result[:response]
+      logger.info "#{crd} restore tear_down registered:\n#{origin_crd}"
+    else
+      raise "Could not get #{crd}: #{name}"
+    end
+
+    _admin = admin
+    teardown_add {
+      admin.cli_exec(:delete, object_type: crd, object_name_or_id: name)
+
+      @result = _admin.cli_exec(:create, f: "-", _stdin: origin_crd)
+      raise "Cannot restore #{crd}: #{name}" unless @result[:success]
+    }
+  else
+    raise "No such #{crd} CRD in the cluster"
```

Comment:
error message better be somehting like: #{crd} not supported by this step, or something along these lines.